runtime/turing/src/migrations/register_shiden_asset.rs

```diff
@@ -0,0 +1,119 @@
+use crate::*;
+use frame_support::{pallet_prelude::Weight, traits::OnRuntimeUpgrade};
+use orml_asset_registry::AssetMetadata;
+use primitives::assets::CustomMetadata;
+
+pub type AssetMetadataOf = AssetMetadata<Balance, CustomMetadata>;
+
+pub fn dollar(decimals: u32) -> Balance {
+	10_u128.pow(decimals)
+}
+
+pub fn cent(decimals: u32) -> Balance {
+	dollar(decimals) / 100
+}
+
+#[frame_support::storage_alias]
+type LastAssetId<T: Config> =
+	StorageValue<AssetRegistry, <T as orml_asset_registry::Config>::AssetId>;
+
+pub struct AddShidenAsset;
+impl OnRuntimeUpgrade for AddShidenAsset {
+	fn on_runtime_upgrade() -> Weight {
+		log::info!(
+			target: "asset_registry",
+			"on_runtime_upgrade: add shiden asset"
+		);
+
+		let asset = AssetMetadataOf {
+			decimals: 18,
+			name: b"Shiden".to_vec(),
+			symbol: b"SDN".to_vec(),
+			additional: CustomMetadata {
+				fee_per_second: Some(416_000_000_000),
```

Comment:
This is a placeholder. We need to decide on a real fee_per_sec before launch.

Reply:
fyi, @chrisli30

Reply:
Let me ask the Astar team about the converstion rate, but let’s think about it, what is the best way to update its value? It doesn’t need to be changed frequently, for example just once a week. Could we write it from a whitelisted wallet address through a RPC?

Reply:
The hard-coded value currently works for this PR and demo purpose.

Reply:
This value can be updated by governance. This brings up something I forgot to note- For turing-staging, we'll need to add the shibuya token with a governance proposal. Adding it here in a migration will also add it to turing since turing and turing-staging share one runtime.

Reply:
okay, got it yeah as long as it’s adjustable with Governance. I’m registering the asset to Turing Staging.